.github/workflows/dependabot.yaml

```diff
@@ -0,0 +1,91 @@
+name: dependabot
+
+on:
+  pull_request:
+    types:
+      - opened
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+# Only run one instance per PR to ensure in-order execution.
+concurrency: pr-${{ github.ref }}
+
+jobs:
+  # Dependabot is annoying, but this makes it a bit less so.
+  dependabot-automerge:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.event.pull_request.user.login == 'dependabot[bot]' && github.repository == 'coder/coder'
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@d7267f607e9d3fb96fc2fbe83e0af444713e90b7 # v2.3.0
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Approve the PR
+        run: gh pr review --approve "$PR_URL"
```

Comment:
Should we include a message here saying it was auto-approved?

Reply:
A message like a comment on the PR? I think it's not needed as the GitHub UI is self explanatory and it shows the PR is approved by a bot. 
We are additionally sending notifications for these too.

Reply:
I can add an `echo` statement here to print this ti logs.